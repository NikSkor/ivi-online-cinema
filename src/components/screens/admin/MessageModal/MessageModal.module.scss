@import '../../../../styles/vars.scss';

.modal {
  height: 100vh;
  width: 100vw;
  background-color: rgba(0,0,0,0.3);
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
  opacity: 0;
  transition: all 0.5s;
  pointer-events: none;
}

.modal.active {
  opacity: 1;
  pointer-events: all;
}

.modalContent {
  padding: 20px;
  border-radius: 15px;
  background-color: $lightGray;
  width: 400px;
  min-height: 200px;
  transform: scale(0.5);
  transition: all 0.5s;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-direction: column;
  gap: 20px;
}

.modalContent.active {
  transform: scale(1);
}

.modalText {
  font-size: 24px;
  line-height: 26px;
  font-weight: 500;
  font-style: normal;
  text-align: center;
}

.btnClose {

  font-weight: 500;
  font-size: 14px;

  position: relative;
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  height: 40px;
  padding: 12px 12px;

  border: solid 1px $btn_bg;
  background-color: $btn_bg;
  border-radius: 8px;

  cursor: pointer;
  text-decoration: none;
  color: $primary;

  transition-property: border-color, background-color;
  transition-duration: .4s;

  &:not(:last-child) {
    margin-right: 12px;
  }

  &.square {
    width: 40px;
    padding: 0;
  }

  &:hover {
    background-color: $btn_hover;
    border-color: $btn_hover;
  }
}