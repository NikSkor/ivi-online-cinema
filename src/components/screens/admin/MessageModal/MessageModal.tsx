import { FC, useState } from "react";
import style from './MessageModal.module.scss';
import Link from "next/link";

interface IMessageModal {
  active: boolean,
  setActive: Function,
  message: string,
  link?: string,
  setValidateName?: Function
}

const MessageModal: FC<IMessageModal> = ({active, setActive, message, link, setValidateName}) => {

  
  return (
    <div className={active ? `${style.modal} ${style.active}` : style.modal} onClick={(e)=> e.stopPropagation()}>
      <div className={active ? `${style.modalContent} ${style.active}` : style.modalContent} onClick={(e)=> e.stopPropagation()}>
        <p className={style.modalText}>{message}</p>
        {link
          ? 
          <Link href={link}>
            <button 
              className={style.btnClose} 
              onClick={()=> {
                setActive(false);
              }}>Закрыть</button>
          </Link>
          :
          <button 
              className={style.btnClose} 
              onClick={()=> {
                setActive(false);
                if (typeof setValidateName === 'function') {
                  setValidateName(true);
                }
              }}>Закрыть</button>
        }
      </div>
    </div>
  )
}

export default MessageModal;