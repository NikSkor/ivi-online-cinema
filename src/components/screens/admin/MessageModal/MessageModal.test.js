import MessageModal from '@/components/screens/admin/MessageModal/MessageModal';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import mockRouter from 'next-router-mock';

jest.mock('next/router', () => require('next-router-mock'));


describe('test MessageModal component', () => {
  afterEach(cleanup);

  test('there is the button', () => {
    render(
      <MessageModal
        active={true}
        setActive={(e) => {}}
        link={'/admin'}
        message={'Всё работает!'}
        isValidName={true}
      ></MessageModal>
    );
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('extBtn returns to admin page', () => {
      mockRouter.push('/initial-path');

      render(
        <MessageModal
          active={true}
          setActive={(e) => {}}
          link={'/foo?bar=baz'}
          message={'Всё работает!'}
          isValidName={true}
        ></MessageModal>
      );

      fireEvent.click(screen.getByRole('button'));

      expect(mockRouter).toMatchObject({
        asPath: '/foo?bar=baz',
        pathname: '/foo',
        query: { bar: 'baz' },
      });
  });

  test('component renders props correctly, truthy values', () => {
    const tree = renderer
      .create(
        <MessageModal
          active={true}
          setActive={(e) => {}}
          link={'/foo?bar=baz'}
          message={'Всё работает!'}
          isValidName={true}
        ></MessageModal>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderer.create(<MessageModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});