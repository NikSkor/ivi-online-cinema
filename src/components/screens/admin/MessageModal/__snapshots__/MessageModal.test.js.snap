// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Modal Window component there is the button 1`] = `
<button
  class="btnClose"
>
  Закрыть
</button>
`;
